services/cd-service/pkg/service/git.go

```diff
@@ -153,10 +157,95 @@ func (s *GitServer) GetCommitInfo(ctx context.Context, in *api.GetCommitInfoRequ
 	for _, dir := range dirs {
 		touchedApps = append(touchedApps, dir.Name())
 	}
-	sort.Strings(touchedApps);
+	sort.Strings(touchedApps)
+
+	events, err := s.GetEvents(ctx, fs, in.CommitHash, commitPath)
+	if err != nil {
+		return nil, err
+	}
 
 	return &api.GetCommitInfoResponse{
 		CommitMessage: commitMessage,
-		TouchedApps: touchedApps,
+		TouchedApps:   touchedApps,
+		Events:        events,
 	}, nil
-}
\ No newline at end of file
+}
+
+func (s *GitServer) GetEvents(ctx context.Context, fs billy.Filesystem, hash string, commitPath string) ([]*api.Event, error) {
+	errorTemplate := func(message string, err error) error {
+		return fmt.Errorf("could not get events for hash '%s' in path '%s' - error %w", hash, commitPath, err)
+	}
+	var result []*api.Event = nil
+	allEventsPath := fs.Join(commitPath, "events")
+	potentialEventDirs, err := fs.ReadDir(allEventsPath)
+	if err != nil {
+		return nil, errorTemplate("could not read events directory ", err)
+	}
+	for i := range potentialEventDirs {
+		oneEventDir := potentialEventDirs[i]
+		if oneEventDir.IsDir() {
+			fileName := oneEventDir.Name()
+			rawUUID, err := timeuuid.ParseUUID(fileName)
+			if err != nil {
+				return nil, errorTemplate(fmt.Sprintf("could not read event, because it is not a UUID '%s'", fileName), err)
+			}
+
+			var event *api.Event = nil
+			event, err = s.ReadEvent(ctx, fs, fs.Join(allEventsPath, oneEventDir.Name()), rawUUID)
```

Comment:
Hm, to me it would make more sense that the events are stored in `events/<event-UUID>` instead of `events/<event-name>/<event-UUID>`. What is the benefit of specifying the event name?